src/opnsense/www/js/opnsense.js

```diff
@@ -120,32 +125,32 @@ function getFormData(parent) {
  */
 function setFormData(parent,data) {
     $( "#"+parent+"  input,#"+parent+" select,#"+parent+" span,#"+parent+" textarea" ).each(function( index ) {
-        if ($(this).prop('id') == undefined || $(this).prop('id') == "") {
+        if ($(this).prop('id') === undefined || $(this).prop('id') === "") {
             // we need an id.
             return;
         }
         var node = data;
         var targetNode = $(this); // document node to fetch data to
         var keyparts = $(this).prop('id').split('.');
-        $.each(keyparts,function(indx,keypart){
+        $.each(keyparts,function(index, keypart){
             if (keypart in node) {
-                if (indx < keyparts.length - 1) {
+                if (index < keyparts.length - 1) {
                     node = node[keypart];
                 } else {
                     // data node found, handle per type
                     if (targetNode.is("select")) {
                         // handle select boxes
                         targetNode.empty(); // flush
                         $.each(node[keypart],function(indxItem, keyItem){
-                            var opt = $("<option>").val(indxItem).text(keyItem["value"]);
-                            if (keyItem["selected"] != "0") {
+                            var opt = $("<option>").val(htmlDecode(indxItem)).text(keyItem["value"]);
+                            if (+keyItem["selected"] !== 0) {
                                 opt.attr('selected', 'selected');
                             }
                             targetNode.append(opt);
                         });
-                    } else if (targetNode.prop("type") == "checkbox") {
+                    } else if (targetNode.prop("type") === "checkbox") {
                         // checkbox type
-                        targetNode.prop("checked", node[keypart] != 0);
+                        targetNode.prop("checked", +node[keypart] !== 0);
```

Comment:
`+`node[keypart] ? why the [+]? shouldn't this compare to `"0"` in your case?

Reply:
+ is a shortcut for convert to a numeric value so it compares 0!==0 instead of "0"!==0.

this prevents some questionable true like comparisons:

```js
""==0
true

+""==0
true

+"a"==0 // the value left is NaN in this case
false
```